gpt_index/indices/response/schema.py

```diff
@@ -46,11 +47,19 @@ def __str__(self) -> str:
         """Convert to string representation."""
         return self.response or "None"
 
-    def get_formatted_sources(self) -> str:
+    def get_formatted_sources(self, documents: Optional[List[Document]] = None) -> str:
         """Get formatted sources text."""
         texts = []
         for source_node in self.source_nodes:
             fmt_text_chunk = truncate_text(source_node.source_text, 100)
             doc_id = source_node.doc_id or "None"
-            texts.append(f">Source (Doc id: {doc_id}): {fmt_text_chunk}")
+            source_text = f"> Source (Doc id: {doc_id}): {fmt_text_chunk}"
+
+            # If `documents` exists, add some helpful info about the document
```

Comment:
I would leave this part out for now for a few reasons:
- the source_node.source_text would already include the extra_info from the node if it was specified
- since we return the doc_id, the user could look up the source document (with extra_info) if they wanted to
- passing in a list of documents into get_formatted_sources isn't the most intuitive function interface

Reply:
Ah yes, I initially added that in when I was just adding metadata to the document and not all the nodes (which I think there's still a use case for, but it's not super relevant anymore in the demo I'm working on). 

Deleted.